.github/workflows/continous-integration.yml

```diff
@@ -139,13 +139,34 @@ jobs:
       with:
         poetry-version: ${{ env.POETRY_VERSION }}
 
-    - name: Load Poetry Cached Libraries ⬇
+    - name: Set up virtual environment (Windows)
+      if: matrix.os == 'windows-latest'
+      # Poetry on Windows cannot pick up the virtual environments directory properly,
+      # and it creates a new one every time the pipeline runs.
+      # We want to  allow Poetry to create a venv directory if it doesn't
+      # exist, but also pick it up if it's already there.
+      # This step solves this problem — it tells poetry to always use `.venv` directory inside
+      # the project itself, which also makes it easier for us to determine the correct directory
+      # that needs to be cached.
+      run: |
+        poetry config virtualenvs.in-project true
```

Comment:
@alwx @wochinge wondering if we should do this for both ubuntu and windows, for the sake of keeping it simple 🤔